talk_sample/src/main/java/com/zendesk/talk/sample/LineStatusChecker.kt

```diff
@@ -0,0 +1,35 @@
+package com.zendesk.talk.sample
+
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import zendesk.talk.android.LineStatusResult
+import java.util.concurrent.TimeUnit
+
+class LineStatusChecker {
+
+    private companion object {
+        val LINE_STATUS_CHECK_PERIOD = TimeUnit.MINUTES.toMillis(5)
+    }
+
+    /**
+     * Periodically checks whether the line status is enabled and accessible
+     */
+    fun lineStatus(): Flow<Boolean> {
```

Comment:
Have you thought about triggering the line status check via an additional button instead of polling? I know that given the lack of push notifications, periodically calling line status may be a viable solution in some (perhaps even many) use cases, but I have reservations about showing such a pattern in an official sample app.

Reply:
Good point about polling. I can do the button for this purpose. However I believe that still most of integrators would need to implement the polling mechanism anyway. 

However I agree that showing that in official sample app would result in having the polling implemented way often then needed.